### PR TITLE
Fixes issues in PR#2

### PR DESCRIPTION
**Pull request by michellemtchai**
- This is based off @simpslandyy's PR #2
- This PR target PR #2 for merging, **not** master
- **Please await @simpslandyy's approval before trying to merge this.**
- This fixed the `Connection Refused` issue as well as the `Open Timeout`
- This currently only works for both Gmail and Outlook
- Steps to get this working:
  - run `cp dev.env.default .env`
  - set the variables `EMAIL_TYPE`, `EMAIL_USER`, `EMAIL_PASSWORD`, and `EMAIL_AUTH` in `.env`  from the pinned info on Slack
    - if you want to use your own email, follow the instruction near the bottom of the description to make sure your email is authenticated to send messages using the CP  app
  - modify line of `hello_react_controller` and change my email to yours
  - open up the email you wrote in `hello_react_controller`
  - run `docker-compose up` as usual
  - go to `localhost:5000/hello_react`
  - if it renders a JSON of 2 different offers, then you should get your email within the next minute

Cause of the errors in PR#2:
- the environmental variables `ENV['hotmail_user']` and `ENV['hotmail_pass']` were set to blank because they were never defined. And since they were meant to be the sender. PR #2 was basically trying to send emails anonymously.

Setting up your own account as the sender (Extra):
- Gmail:
  - sign into your Gmail account
  - go to [link 1](https://myaccount.google.com/lesssecureapps) and [link 2](https://accounts.google.com/DisplayUnlockCaptcha)
    - you have to turn off the security and display unlock captcha
- Outlook:
  - download the Outlook (2016) Desktop app onto your PC
  - Start up Outlook, go to `File` in the top menu
  - Go to `Info` on the side bar, and click `Account Settings` on the right panel
  - Click on the drop down that appears after you click on the button
  - In the pop-up window, stay on the `Email` tab
  - Click `New...` and choose `Manual setup or additional server types`
  - On the next screen, choose `POP or IMAP`
  - This next screen requires information you can only get from your email account, so you have to log into the Outlook email you want to configure as host
  - In your online email account, click on the gear button to the top right and choose `Options`
  - Then, on the sidebar, go to `Mail > Accounts > POP and IMAP`
  - The right panel that appears after you clicked on the right link on the sidebar will give you the necessary information to fill in the form in your Outlook app setup
  - For the following fields, enter your Outlook email:
    - `Your Name`
    - `Email Address`
    - `User Name`
  - Enter the password to this email in `Password`
  - Choose the `Account Type` to be IMAP
  - set the `Incoming mail server` as the `server name` under `IMAP setting` on in your email account
  - set the `Outgoing mail server (SMTP)` as the `server name` under `SMTP setting` on in your email account
  - Then click on the button `More Settings...` on the bottom right corner of the dialog 
  - A new pop-up should appear
  - Go to the `Outgoing Server` tab, click on the checkbox `My outgoing server (SMTP) requires authentication`
  - Choose the radio button `Use the same settings as my incoming mail server`
  - Then, move onto the next tab `Advanced`
  - Choose the type of encryption for both the incoming/outgoing server as `TLS`
  - Click `ok` on the bottom of this dialog
  - Click `Next >` on the remaining dialog, and once the test that Outlook run both pass, and you're done!

### CODE REVIEW

File: app/controllers/hello_react_controller.rb [hunk 1]
```diff
@@ -2,6 +2,13 @@ class HelloReactController < ApplicationController
   def contracts
     @contractor = Contractor.new("seeds/offers.json")
     @contractor.build_contracts
+    @user = {
+      name: "Landy Simpson",
+      course: "CSC104H1F",
+      filename: "#{Rails.root}/tmp/contracts/1342088432-CSC104H1F-contract.pdf",
+      email: "mingtsengeh@gmail.com"
+      }
+    test = CpMailer.contract_email(@user).deliver_now!
```
Comment (freeatnet) [inline app/controllers/hello_react_controller.rb]: [`deliver_now!`](http://api.rubyonrails.org/classes/ActionMailer/MessageDelivery.html#method-i-deliver_now-21) seems to be a pretty harsh method:
> Delivers an email without checking perform_deliveries and raise_delivery_errors, so use with caution.

we should probably use `deliver_now` instead.

Comment (michellemtchai) [inline app/controllers/hello_react_controller.rb]: Okay. I'll change that.

**Comment by simpslandyy** [comment]
@michellemtchai I know you're excited about this, but please try to understand that you're now doing my task, and some of these fixes are in my branch :( So now we have two branches attempting to fix the same problem. If you would like like to take on the mailer task then I will discard my changes from my branch.

**Comment by michellemtchai** [comment]
@simpslandyy I'm really sorry for barging into your task like this. Your description on PR #2 looked distressful, so I thought I could help out a bit since I have the time. This PR is actually targeted for your branch; I actually only meant to make a few small changes that may take away your blockade, and let you merge it back to your branch. I didn't realize that you actually have other commits locally. I really hope this won't impede your workflow as it was meant to do the opposite.

**Comment by simpslandyy** [comment]
That's okay just talk to me first, that way I can communicate the problem I'm having and the steps I've taken and most importantly explain what I'm doing.

On Jul 22, 2017 11:01 AM, michellemtchai <notifications@github.com> wrote:

@simpslandyy<https://github.com/simpslandyy> I'm really sorry for barging into your task like this. Your description on PR #2<https://github.com/uoft-tapp/cp/pull/2> looked distressful, so I thought I could help out a bit since I have the time. This PR is actually targeted for your branch; I actually only meant to make a few small changes that may take away your blockade, and let you merge it back to your branch. I didn't realize that you actually have other commits locally. I really hope this won't impede your workflow as it was meant to do the opposite.

**Comment by michellemtchai** [comment]
@simpslandyy Okay. I'll definitely ask you before I make any assumption from now on.

**Pull request merged**